### ISSUE TITLE
Can not join network by netclient if interfaces does not have any mac address

### ISSUE THREAD

**Issue by Hintay** [gravitl/netmaker#733]
I'm trying join network by netclient on a OpenVZ container. If the network interfaces do not have any mac addresses present, the following error will occur.

```
2022/02/10 19:32:41 could not retrieve mac address
```

At the same time, netclient assumes that it has joined the network without having successfully joined it. Re-run the join command and the following result will appear.
```
2022/02/10 19:40:00 [netclient] success
2022/02/10 19:40:00 [netclient] ALREADY_INSTALLED. Netclient appears to already be installed for default. To re-install, please remove by executing 'sudo netclient leave -n default'. Then re-run the install command.
```
Here is what it shows when I run the leave network command.
```
# netclient leave -n default
2022/02/10 19:43:45 Failed to authenticate: rpc error: code = Unauthenticated desc = Something went wrong with Auto Login: rpc error: code = Unknown desc = missing mac address
2022/02/10 19:43:45 [netclient] removed default network locally
2022/02/10 19:43:45 [netclient] removed systemd remnants if any existed
2022/02/10 19:43:45 [netclient] success
```

The following should be the code to get the mac address, it may be that using a stored random mac on a no mac address system would be a good option.
https://github.com/gravitl/netmaker/blob/93978c171a2f769318c1d2c45ec372873dfbd48c/netclient/ncutils/netclientutils.go#L171-L185

**Comment by mattkasun** [comment]
you can specify a macaddress as a parameter to the join command 
`netclient join -macaddress 00:01:02:03:04:05:06`

**Comment by afeiszli** [comment]
fixed in v0.11.0, macaddress is no longer mandatory
